gdnative-derive/src/native_script/property_args.rs

```diff
@@ -37,6 +43,57 @@ impl<'a> Extend<&'a syn::MetaNameValue> for PropertyAttrArgsBuilder {
                         panic!("there is already a path set: {:?}", old);
                     }
                 }
+                "get_hook" => {
+                    let string = if let syn::Lit::Str(lit_str) = &pair.lit {
+                        lit_str.value()
+                    } else {
+                        panic!("get_hook value is not a string literal");
+                    };
+
+                    if let syn::Expr::Path(path_expr) =
+                        syn::parse_str::<syn::Expr>(string.as_str()).expect("Invalid syntax.")
+                    {
+                        if let Some(old) = self.get_hook.replace(path_expr.path) {
+                            panic!("there is already a get_hook value set: {:?}", old);
+                        }
+                    } else {
+                        panic!("Not a path expression.")
```

Comment:
Why parse as a path expression? `syn::Path` itself also implements `Parse`. You can just parse it directly.

Reply:
I am working on this now. My theory was that having the expression might allow better error messages. However, I can't really return errors from the extend method, so I guess that doesn't matter.

FWIW, this is probably something that should be fixed in the future. Implementing the Extend trait doesn't make a ton of sense to me since it doesn't seem to give us anything here. And it would be super useful to be able to return errors from that function so that we could have better compiler error messages. For example, when I was getting the ExprPath instead, it has all the spans that I could use to generate a good syn:Error so that the compiler could show the bad code instead of only the macro invocation, which is harder than it needs to be to debug.

Reply:
Okay. I have addressed this.